apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -262,23 +263,41 @@ Value CreateStructValue(IEnumerable<KeyValuePair<string, object>> fields)
             }
         }
 
-        /// <summary>
-        /// Gets the full trace name if the log target is a project, we have an
-        /// HTTP accessor and a valid trace header exists on the current context.
-        /// If the trace name cannot be determined null is returned.
-        /// See: See: https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
-        /// </summary>
-        internal string GetTraceName()
+        private void SetTraceAndSpanIfAny(LogEntry entry)
         {
-            var httpContext = _serviceProvider?.GetService<IHttpContextAccessor>()?.HttpContext;
-            if (_traceTarget == null || httpContext == null)
+            if (_traceTarget is null)
+            {
+                return;
+            }
+
+            string traceId = null;
+            ulong? spanId = null;
+
+            // If there's currently a Google trace and span use that one.
+            // This means that the Google Trace component of the diagnostics library
+            // has been initialized.
+            if(ContextTracerManager.GetCurrentTracer() is IManagedTracer tracer && tracer.GetCurrentTraceId() is string googleTraceId)
+            {
+                traceId = googleTraceId;
+                spanId = tracer.GetCurrentSpanId();
+            }
+            // Else let's look at the header.
+            else if (_serviceProvider?.GetService<IHttpContextAccessor>()?.HttpContext is HttpContext httpContext)
+            {
+                var traceContext = TraceHeaderContext.FromHeader(
+                    httpContext.Request?.Headers[TraceHeaderContext.TraceHeader]);
```

Comment:
```suggestion
                var traceContext = TraceHeaderContext.FromHeader(httpContext.Request?.Headers[TraceHeaderContext.TraceHeader]);
```

Reply:
It's not much longer than a few lines earlier, and easier to read. Or maybe break it up:

```csharp
var header = httpContext.Request?.Headers[TraceHeaderContext.TraceHeader];
var traceContext = TraceHeaderContext.FromHeader(header);
```

I'm fine with either option.